packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -206,13 +206,16 @@ class RenderAndroidView extends RenderBox with _PlatformViewGestureMixin {
     // Clip the texture if it's going to paint out of the bounds of the renter box
     // (see comment in _paintTexture for an explanation of when this happens).
     if (size.width < _currentAndroidViewSize.width || size.height < _currentAndroidViewSize.height && clipBehavior != Clip.none) {
-      context.pushClipRect(true, offset, offset & size, _paintTexture, clipBehavior: clipBehavior);
+      _clipRectLayer = context.pushClipRect(true, offset, offset & size, _paintTexture, clipBehavior: clipBehavior,
+           oldLayer: _clipRectLayer);
       return;
     }
-
+    _clipRectLayer = null;
```

Comment:
This line looks like it should be inside an else clause, but it is not.

Reply:
Because the block before ends with return. But I can put it in else clause.

Reply:
Ah, I didn't see that. I didn't expect a return there since the method isn't done and it almost looked like a bug, but I see now that the pushCR call invokes the _paintTexture method as a callback.